nodewatcher/modules/monitor/http/survey/processors.py

```diff
@@ -0,0 +1,106 @@
+
+
+class SurveyInfo(monitor_processors.NodeProcessor):
+    """
+    Accesses the data collected during a WiFi survey of neighboring wireless access points / routers. Stores channel,
+    BSSID, SSID and signal strength for each neighbor into the datastream. A star shaped graph is constructed. The
+    source vertex contains an array of all BSSIDs of that node. Source vertex is stored according to its UUID,
+    wherease all neighbors are stored according to their BSSIDs. Will only run if HTTP monitor module has previously
+    fetched data.
+    """
+
+    @monitor_processors.depends_on_context("http", http_processors.HTTPTelemetryContext)
+    def process(self, context, node):
+        """
+        Called for every processed node.
+
+        :param context: Current context
+        :param node: Node that is being processed
+        :return: A (possibly) modified context
+        """
+
+        version = context.http.get_module_version('core.wireless')
+        if version == 0:
+            # Unsupported version or data fetch failed
+            return context
+
+        edges, vertices = [], []
+        try:
+            source_vertex_bssids = [interface.bssid for interface in context.http.core.wireless.interfaces.values()]
+            source_vertex = {
+                'i': str(node.uuid),
+                'bssids': source_vertex_bssids,
+            }
+            vertices.append(source_vertex)
+            for radio in context.http.core.wireless.radios.values():
+                for neighbor in radio.survey:
+                    vertices.append({'i': neighbor['bssid']})
+                    # Add an edge for that neighbor.
+                    edge = {'f': str(node.uuid), 't': neighbor['bssid']}
+                    for field in ('channel', 'signal', 'ssid'):
+                        if field in neighbor:
+                            edge[field] = neighbor[field]
+                    edges.append(edge)
+        except KeyError:
+            pass
+        except NameError:
+            self.logger.warning("Could not parse survey data for node: " + str(node))
+
+        latest_stored_graph = None
+        # Retrieve the latest stored datapoint, if it exists
+        streams = datastream.find_streams({'node': node.uuid})
+        if streams:
+            latest_stream = streams[len(streams) - 1]
+            datapoint_iterator = datastream.get_data(stream_id=latest_stream['stream_id'],
+                                                     granularity=latest_stream['highest_granularity'],
+                                                     start=latest_stream['latest_datapoint'])
+            try:
+                latest_stored_graph = datapoint_iterator[0]['v']
+            except IndexError:
+                pass
+
+        latest_graph = {
+            "v": vertices,
+            "e": edges
+        }
+
+        if latest_graph != latest_stored_graph:
+            # Store the latest graph into datastream.
```

Comment:
We use timestamps with timezones. If timezone is not available, [UTC timezone is set](https://github.com/wlanslovenija/datastream/blob/master/datastream/api.py#L442).

I think `latest_datapoint` contains a timezone as well. So it does not matter really how you obtain your time, you can just compare timestamps. Python has much better datetime support than JavaScript. So converting to timezones in general is not so important. The only issue is if you do or not use timezones.

You can read more about timezones in Python if you care.

In Python, the best is to use [now](https://github.com/django/django/blob/master/django/utils/timezone.py#L320) function provided by Django. And then just compare with latest datapoint.